src/features/lightspeed/inlineSuggestions.ts

```diff
@@ -835,5 +823,61 @@ export async function inlineSuggestionUserActionHandler(
     inlineSuggestionFeedbackPayload,
     lightSpeedManager.orgTelemetryOptOut
   );
+  resetSuggestionData();
+}
+
+function inlineSuggestionPending(): boolean {
+  if (vscode.window.activeTextEditor?.document.languageId !== "ansible") {
+    return false;
+  }
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    return false;
+  }
+  if (!inlineSuggestionData["suggestionId"]) {
+    return false;
+  }
+  return true;
+}
+
+function resetSuggestionData(): void {
   inlineSuggestionData = {};
+  insertTexts = [];
+}
+
+export async function textDocumentChangeHandler(
+  e: vscode.TextDocumentChangeEvent
+) {
+  // Exit early if Lightspeed or Inline suggestion is not enabled.
+  const lightSpeedSetting =
+    lightSpeedManager.settingsManager.settings.lightSpeedService;
+  if (!lightSpeedSetting.enabled || !lightSpeedSetting.suggestions.enabled) {
+    return;
+  }
+
+  // If the user accepted a suggestion on the widget, ansible.lightspeed.inlineSuggest.accept
+  // command is not sent. This method checks if a text change that matches to the current
+  // suggestion was found. If such a change was detected, we assume that the user accepted
+  // the suggestion on the widget.
+  if (
+    inlineSuggestionPending() &&
+    insertTexts &&
+    e.document.languageId === "ansible" &&
+    e.contentChanges.length > 0
+  ) {
+    const suggestionId = inlineSuggestionData["suggestionId"];
+
+    e.contentChanges.forEach(async (c) => {
+      if (c.text === insertTexts[0]) {
+        // If a matching change was found, send a feedback with the ACCEPTED user action.
+        console.log(
+          "[inline-suggestions] Detected a text change that matches to the current suggestion."
+        );
+        await inlineSuggestionUserActionHandler(
```

Comment:
If user accepts suggestion by TAB key will this method be called again? In the `inlineSuggestionCommitHandler` we already send a accept feedback. If the `textDocumentChangeHandler` is invoked after TAB key press as well in that case we are duplicate the accept feedback?

Reply:
Got it